### PR TITLE
Add base charm structure

### PR DESCRIPTION
**Pull request by rgildein**
- removed all k8s and pebble parts (machine charm)
 - prepared metadata.yaml file (relations, resources, ...)
 - wrote README.md
 - added installation hook that installs prometheus-bind-exporter snap

### CODE REVIEW

File: metadata.yaml [hunk 1]
```diff
@@ -1,19 +1,31 @@
 # Copyright 2021 Unicorn
 # See LICENSE file for licensing details.
-name: prometheus-bind-exporter-operator
+name: prometheus-bind-exporter
 description: |
-  TEMPLATE-TODO: fill out the charm's description
+  The subordinate charm for the designate-bind charm, which provide the
```
Comment (mkalcok) [inline metadata.yaml]: which provide**s**

File: README.md [hunk 2]
```diff
@@ -2,11 +2,29 @@
 
 ## Description
 
-TODO: Describe your charm in a few paragraphs of Markdown
+The subordinate charm for the designate-bind charm, which provide the
```
Comment (mkalcok) [inline README.md]: which provide**s**

File: actions.yaml [hunk 3]
```diff
@@ -1,16 +1,2 @@
 # Copyright 2021 Unicorn
```
Comment (mkalcok) [inline actions.yaml]: If the charm does not have any actions, this file can be removed completely.

File: config.yaml [hunk 4]
```diff
@@ -1,14 +1,4 @@
 # Copyright 2021 Unicorn
```
Comment (mkalcok) [inline config.yaml:1]: maybe this file can be removed too, if there are no config options? Not sure about it but you can give it a try.

File: metadata.yaml [hunk 5]
```diff
@@ -1,19 +1,31 @@
...
 resources:
-  httpbin-image:
-    type: oci-image
-    description: OCI image for httpbin (kennethreitz/httpbin)
+  prometheus-bind-exporter:
+    type: file
+    filename: promethues-bind.exporter.snap
 
+requires:
+  juju-info:
```
Comment (mkalcok) [inline metadata.yaml:20]: This is just a suggestion but I recently also made a new subordinate charm taht's meant to be deployed on `kubernetes-master` and I named the subordinate relation `kubernetes` just to give a bit of a context to the user

**Review by mkalcok — CHANGES_REQUESTED** [review]
1 typo and few cosmetic suggestions, otherwise LGTM.

File: actions.yaml (hunk 3 above)
Comment (rgildein) [reply actions.yaml]: Good point, I don't think there will be any action in the near future.

File: config.yaml (hunk 4 above)
Comment (rgildein) [reply config.yaml:1]: There could be others here, because I'm already starting to work on this (there will be a new PR).

**Review by mkalcok — APPROVED** [review]

**Pull request merged**